vividus-plugin-rest-api/src/main/java/org/vividus/http/PublishingAttachmentInterceptor.java

```diff
@@ -63,6 +67,10 @@ public void process(HttpRequest request, EntityDetails entityDetails, HttpContex
                                 Optional.ofNullable(ContentType.parseLenient(entity.getContentType()))
                                         .orElse(ContentType.DEFAULT_TEXT).getMimeType()
                         );
+                if (entity instanceof ByteArrayEntity)
+                {
+                    charset = StandardCharsets.US_ASCII;
```

Comment:
what is the logic behind it?

Reply:
as I understand, the one ability to have binary content in body - using anything like
```
Given request body: #{loadBinaryResource(/files/surgeon_testPhoto.jpg)}
When I execute HTTP POST request for resource with relative URL `/upload/png`
```
in such case we don't have any header to define that is binary content only entity is instanceof `ByteArrayEntity`

Reply:
that's true, you can:
- check content-type header (see logic above)
- rely on type of entity as a fallback

but my question is why do you set this charset? what is the result?